src/buildSummary.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright 2024 The MathWorks, Inc.
+import * as core from "@actions/core";
+import { join } from 'path';
+import { readFile } from 'fs';
+import { promisify } from 'util';
+
+// Promisify the readFile function to use it with async/await
+const readFileAsync = promisify(readFile);
+
+export interface Task {
+  name: string;
+  description: string;
+  failed: boolean;
+  skipped: boolean;
+  duration: string;
+}
+
+export interface TaskList {
+  taskDetails: Task[];
+}
+
+export async function readJsonFile(filePath: string): Promise<TaskList> {
+  try {
+    const data = await readFileAsync(filePath, { encoding: 'utf8' });
+    return JSON.parse(data);
+  } catch (error) {
+    console.error('Error reading the JSON file:', error);
+    throw error;
+  }
+}
+
+export function getBuildSummaryTable(tasks: TaskList): string[][] {
+  const header: string[] = ['MATLAB Build Task', 'Status', 'Description', 'Duration (HH:MM:SS)'];
+  let taskSummaryTableRows: string[][] = [header];
+
+  tasks.taskDetails.forEach((task, index) => {
+    let taskDetails: string[] = [];
+    taskDetails.push(task.name);
+    if (task.failed) {
+      taskDetails.push('🔴 FAILED');
+    } else if (task.skipped) {
+      taskDetails.push('🔵 SKIPPED');
+    } else {
+      taskDetails.push('🟢 SUCCESS');
+    }
+    taskDetails.push(task.description);
+    taskDetails.push(task.duration);
+
+    taskSummaryTableRows.push(taskDetails);
+  });
+
+  return taskSummaryTableRows;
+}
+
+export function writeSummary(taskSummaryTableRows: string[][]) {
+    core.summary
+      .addTable(taskSummaryTableRows)
+      .write();
+}
+
+export async function processAndDisplayBuildSummary() {
+  const runId = process.env.GITHUB_RUN_ID;
+
+  if (!runId) {
+    console.error('GITHUB_RUN_ID environment variable is not set. Unable to locate the build summary file.');
+    return;
+  }
+
+  try {
+    const runnerTemp = process.env.RUNNER_TEMP;
+    const filePath = join(runnerTemp as string, `buildSummary_${runId as string}.json`);
+    const data = await readJsonFile(filePath);
+    const taskSummaryTableRows = getBuildSummaryTable(data);
+    writeSummary(taskSummaryTableRows);
```

Comment:
Can we move the declarations that can't throw outside of the `try` block to make it more obvious what we're potentially catching.

Note: if we refactored slightly like the comment below suggests, I don't think anything within this `try` block would be able to throw anymore, so we could maybe remove it to simplify the logic slightly.

Reply:
I think same as above we need to handle if those two lines failed.